protocol/consistencychecks.go

```diff
@@ -245,7 +245,11 @@ func (cc *ConsistencyChecks) updateTBs(requestType int, msg *Response,
 	case RegistrationType:
 		df := msg.DirectoryResponse.(*DirectoryProof)
 		if df.AP.ProofType() == m.ProofOfAbsence {
-			if err := cc.verifyReturnedPromise(df, key); err != nil {
+			tbKey := key
+			if msg.Error == ReqNameExisted {
+				tbKey = df.TB.Value
+			}
```

Comment:
> although the response is consistent in and of itself, it isn't consistent with our expectation that the already registered key is the one we've supplied.

I'm not sure if we are in the same page, but this is the reason why I want to this fail later. I think most of errors could be use for whistleblowing, but `ErrBindingsDiffer` shouldn't be treated that way. I'm thinking what the client should do when receiving `ErrBindingsDiffer`.